### PR TITLE
Move rocksb to gaia

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: database/src/memory.rs [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+    ops::Bound,
+    sync::{Arc, RwLock},
+};
 
 use crate::Database;
 
 #[derive(Debug, Clone)]
 pub struct MemDB {
-    store: RefCell<BTreeMap<Vec<u8>, Vec<u8>>>, // we use a refcell because the set method on the DB trait doesn't take a mutable ref
+    store: Arc<RwLock<BTreeMap<Vec<u8>, Vec<u8>>>>, // we use a refcell because the set method on the DB trait doesn't take a mutable ref
```
Comment (dmvict) [inline database/src/memory.rs:11]: Update/remove comment, please.

File: gears/src/commands/node/run.rs [hunk 2]
```diff
@@ -32,7 +32,7 @@ pub enum RunError {
     #[error("{0}")]
     HomeDirectory(String),
     #[error("{0}")]
-    Database(#[from] database::error::Error),
+    Database(String),
```
Comment (dmvict) [inline gears/src/commands/node/run.rs:35]: Why did you replaced the type to `String`?

File: database/src/memory.rs (hunk 1 above)
Comment (NYBACHOK) [reply database/src/memory.rs]: Done

File: gears/src/commands/node/run.rs (hunk 2 above)
Comment (NYBACHOK) [reply gears/src/commands/node/run.rs:35]: Error type under `rockdb` feature

File: database/src/lib.rs [hunk 3]
```diff
@@ -23,3 +23,9 @@ pub trait Database {
         prefix: Vec<u8>,
     ) -> Box<dyn Iterator<Item = (Box<[u8]>, Box<[u8]>)> + 'a>;
 }
+
+pub trait DatabaseBuilder<DB> {
```
Comment (joneskm) [inline database/src/lib.rs:27]: Would it make more sense for the DB to be an associated type?

Comment (NYBACHOK) [reply database/src/lib.rs:27]: I thought about it, but think it's easier to use simple generic + generic allows different implementations.
I mean that you could use single structure for different databases.

Comment (joneskm) [reply database/src/lib.rs:27]: Fair enough.

**Pull request merged**